codec/src/main/java/io/netty/handler/codec/MessageToByteEncoder.java

```diff
@@ -46,7 +46,7 @@
 public abstract class MessageToByteEncoder<I> extends ChannelOutboundHandlerAdapter {
 
     private final TypeParameterMatcher matcher;
-    private final boolean preferDirect;
+    protected final boolean preferDirect;
```

Comment:
That won't help in the case of where I'm accessing that value from: `flush()`. We could either create a `protected boolean isPreferDirect()`, or ... ?